internal/driver/luks.go

```diff
@@ -22,16 +22,68 @@ limitations under the License.
 package driver
 
 import (
+	"context"
 	"errors"
 	"fmt"
-	"os/exec"
+	"strconv"
 	"strings"
 
-	mountmanager "github.com/linode/linode-blockstorage-csi-driver/pkg/mount-manager"
 	"k8s.io/klog/v2"
 	utilexec "k8s.io/utils/exec"
+
+	cryptsetup "github.com/martinjungblut/go-cryptsetup"
+
+	cryptsetupclient "github.com/linode/linode-blockstorage-csi-driver/pkg/cryptsetup-client"
+	mountmanager "github.com/linode/linode-blockstorage-csi-driver/pkg/mount-manager"
 )
 
+// CryptSetup manages encrypted devices.
+type CryptSetup struct {
+	_ cryptsetupclient.CryptSetupClient
+}
+
+// Init opens a crypt device by device path.
+func (c CryptSetup) Init(devicePath string) (cryptsetupclient.Device, error) {
+	device, err := cryptsetup.Init(devicePath)
+	if err != nil {
+		return nil, fmt.Errorf("init cryptsetup by device path %q: %w", devicePath, err)
+	}
+	return device, nil
+}
+
+// InitByName opens an active crypt device using its mapped name.
+func (c CryptSetup) InitByName(name string) (cryptsetupclient.Device, error) {
+	device, err := cryptsetup.InitByName(name)
+	if err != nil {
+		return nil, fmt.Errorf("init cryptsetup by name %q: %w", name, err)
+	}
+	return device, nil
+}
+
+type LuksDevice struct {
+	device cryptsetupclient.Device
+}
+
+func NewLuksDevice(crypt cryptsetupclient.CryptSetupClient, path string) (LuksDevice, error) {
+	dev, err := crypt.Init(path)
+	if err != nil {
+		return LuksDevice{}, err
+	}
+	return LuksDevice{device: dev}, nil
+}
+
+func NewLuksDeviceByName(crypt cryptsetupclient.CryptSetupClient, name string) (LuksDevice, error) {
+	dev, err := crypt.InitByName(name)
+	if err != nil {
+		return LuksDevice{}, err
+	}
+	return LuksDevice{device: dev}, nil
+}
+
+func NewCryptSetup() CryptSetup {
+	return CryptSetup{}
+}
+
```

Comment:
question: What do you think about moving this closer to the interface in `pkg/cryptsetup-client`?

Reply:
good call... moved